activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/connector/IntegrationContextBuilder.java

```diff
@@ -53,13 +55,22 @@ private IntegrationContextImpl buildFromExecution(DelegateExecution execution,
         integrationContext.setClientId(execution.getCurrentActivityId());
 
         if(ExecutionEntity.class.isInstance(execution)) {
-            ExecutionEntity processInstance = ExecutionEntity.class.cast(execution)
-                                                                   .getProcessInstance();
+            ExecutionContext executionContext = new ExecutionContext(ExecutionEntity.class.cast(execution));
```

Comment:
@igdianov I was wondering if we shouldn't inject `RepositoryService` here and retrieve the process definition from it. `RepositoryService` implementation is already providing cache via [DeploymentManager](https://github.com/Activiti/Activiti/blob/develop/activiti-engine/src/main/java/org/activiti/engine/impl/persistence/deploy/DeploymentManager.java#L68) and it will be easier to mock than `new ExecutionContext()`. What do you think?

Reply:
@erdemedeiros The ExecutionContext uses ProcessEngineConfiguration to lookup Deployment via ProcessDefinition caching optimization as well as using DeploymentManager entity caching. It think it is better to use existing infrastructure. The mocking is not hard at all.